ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java

```diff
@@ -104,13 +104,32 @@ public OrcSplit(Path path, Object fileId, long offset, long length, String[] hos
     this.isOriginal = isOriginal;
     this.hasBase = hasBase;
     this.rootDir = rootDir;
-    this.deltas.addAll(filterDeltasByBucketId(deltas, AcidUtils.parseBucketId(path)));
+    this.deltas.addAll(filterDeleteDeltasByWriteIds
+            (filterDeltasByBucketId(deltas, AcidUtils.parseBucketId(path)), conf));
     this.projColsUncompressedSize = projectedDataSize <= 0 ? length : projectedDataSize;
     // setting file length to Long.MAX_VALUE will let orc reader read file length from file system
     this.fileLen = fileLen <= 0 ? Long.MAX_VALUE : fileLen;
     this.syntheticAcidProps = syntheticAcidProps;
   }
 
+  /**
+   * For every split we want to filter out the delete deltas that contain events that happened only
+   * in the past relative to the split
+   * @param deltas
+   * @param conf
+   * @return
+   */
+  protected List<AcidInputFormat.DeltaMetaData> filterDeleteDeltasByWriteIds(
+          List<AcidInputFormat.DeltaMetaData> deltas, Configuration conf) throws IOException {
+
+    AcidOutputFormat.Options orcSplitMinMaxWriteIds =
+            AcidUtils.parseBaseOrDeltaBucketFilename(getPath(), conf);
```

Comment:
why not simply `ParsedDeltaLight.parse(bucketFile.getParent())`?

Reply:
done

Reply:
Hi @deniskuzZ,
Many tests failed with the change of using AcidUtils.ParsedDeltaLight.parse() instead of AcidUtils.parseBaseOrDeltaBucketFilename(). As I understand the split is not always a delta folder, it can be some older format not supported by ParsedDeltaLight. I saw that ParsedDeltaLight.parse() is used in some cases internally in AcidUtils.parseBaseOrDeltaBucketFilename(), but not always. Can you please advise if I should revert to using AcidUtils.parseBaseOrDeltaBucketFilename() that worked in all cases or there is some better way?

https://github.com/apache/hive/blob/f6bd0eb80767adfa9ce9f47a6d02a4940903effb/ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java#L538-L552

Reply:
root path could be anything: delta/delete-delta/base. the prev version was actually correct 
````
    long minWriteId = !deltas.isEmpty() ?
      AcidUtils.parseBaseOrDeltaBucketFilename(path, null).getMinimumWriteId() : -1;
    this.deltas.addAll(
      deltas.stream()
        .filter(delta -> isQualifiedDeleteDeltasByWriteIds(delta, minWriteId))
        .flatMap(delta -> filterDeltasByBucketId(delta, bucketId))
        .collect(Collectors.toList()));
````

Reply:
done